### PR TITLE
Implement operator+ for Typedefs of different Tag

### PR DESCRIPTION
**Pull request by akopich**
This would allow for symmetry with `operator-`

Bug: http://b/241776644
Test: Compile, Unit

### CODE REVIEW

File: src/OrbitBase/include/OrbitBase/Typedef.h [hunk 1]
```diff
@@ -135,19 +135,31 @@ class Typedef {
 template <typename Tag>
 class Typedef<Tag, void> {};
 
-// When the `Tag` inherits from the struct, the `Typedef<Tag, T>` supports `operator+`
-struct PlusTag {};
+// When the `Tag` inherits from the struct, the `Typedef<Tag, T>` supports `operator+` with other
+// `Typedef<OtherSummandTag, T>`. The result is wrapped into `Typedef<Tag, T>`.
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/Typedef.h:139]: I think this should mention that if a user doesn't need cross-type summation they can set `OtherSummandTag` to `Tag`.

Nit: In "The result is wrapped into `Typedef<Tag, T>`.", the `T` is in general different to the previous `T`

File: src/OrbitBase/include/OrbitBase/Typedef.h [hunk 2]
```diff
@@ -135,19 +135,31 @@ class Typedef {
...
 };
 
+template <typename First, typename Second, typename FirstDecayed = std::decay_t<First>,
+          typename SecondDecayed = std::decay_t<Second>,
+          typename FirstTag = typename FirstDecayed::Tag,
+          typename SecondTag = typename SecondDecayed::Tag,
+          typename = std::enable_if_t<!std::is_base_of_v<PlusTag<SecondTag>, FirstTag>>>
+auto operator+(First&& lhs, Second&& rhs) {
+  return std::forward<Second>(rhs) + std::forward<First>(lhs);
+};
```
Comment (beckerhe) [inline src/OrbitBase/include/OrbitBase/Typedef.h:161]: Do you also see a way to implement this without assuming commutativity? `operator+(T1, T2)` can theoretically be different to `operator+(T2, T1)`. Of course they shouldn't (in most cases), but even if they weren't (due to a bug), we should call the wrong (the right?) operator here. WDYT?

**Review by beckerhe — APPROVED** [review]
Otherwise LGTM

**Review by florian-kuebler — APPROVED** [review]
I agree with @beckerhe's comments. Beside this, LGTM. Thanks!

File: src/OrbitBase/include/OrbitBase/Typedef.h (hunk 2 above)
Comment (akopich) [reply src/OrbitBase/include/OrbitBase/Typedef.h:161]: You're right. Consider strings for example.

**Pull request merged**